spec/ndx-ophys-devices.extensions.yaml

```diff
@@ -1,8 +1,216 @@
 groups:
-- neurodata_type_def: TetrodeSeries
-  neurodata_type_inc: ElectricalSeries
-  doc: An extension of ElectricalSeries to include the tetrode ID for each time series.
+- neurodata_type_def: Indicator
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Indicator.
   attributes:
-  - name: trode_id
-    dtype: int32
-    doc: The tetrode ID.
+  - name: label
+    dtype: text
+    doc: Indicator standard notation.
+  - name: injection_location
+    dtype: text
+    doc: Injection brain region name.
+    required: false
+  - name: injection_coordinates_in_mm
+    dtype: float
+    shape:
+    - 3
+    doc: Indicator injection location in stereotactic coordinates (AP, ML, DV) mm
+      relative to Bregma.
+    required: false
+- neurodata_type_def: OpticalFiber
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Optical Fiber.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of optical fiber.
+    required: false
+  - name: numerical_aperture
+    dtype: float
+    doc: Numerical aperture, e.g., 0.39 NA.
+    required: false
+  - name: core_diameter_in_um
+    dtype: float
+    doc: Core diameter in micrometers.
+    required: false
+- neurodata_type_def: ExcitationSource
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Excitation Source.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of excitation source device.
+    required: false
+  - name: illumination_type
+    dtype: text
+    doc: 'Type of illumination. Suggested values: LED, LAser or more specifically
+      Gas, Metal vapor, Solid state, Dye, Semiconductor, Free electron, etc.'
+  - name: excitation_wavelength_in_nm
+    dtype: float
+    doc: Excitation wavelength of the stimulation light (nanometers).
+  - name: peak_power_in_W
+    dtype: float
+    doc: Incident power of stimulation device (in Watts).
+    required: false
+  - name: peak_pulse_energy_in_J
+    dtype: float
+    doc: If device is pulsed light source, pulse energy (in Joules).
+    required: false
+  - name: intensity_in_W_per_m2
+    dtype: float
+    doc: Intensity of the excitation in W/m^2, if known.
+    required: false
+  - name: pulse_rate_in_Hz
+    dtype: float
+    doc: If device is pulsed light source, pulse rate (in Hz) used for stimulation.
+    required: false
```

Comment:
Perhaps we've talked about this before - but would it make sense to make a separate class (like `PulsedLightSource`) that requires `pulse_rate_in_Hz`? Or are there generally good reasons to justify keeping a pulsating behavior in all light sources?

Reply:
It's just very common, if it is a Laser excitation source, to be pulsed. I just thought it would be easier to have one device, but if you think it's better to create a separate I can add the `PulsedExcitationSource`

Reply:
I guess it comes down to this

There are two reasons why a field can be optional on a class

i) it always applies but is not often known easily, or is burdensome to uncover or generally uninformative to include

ii) it does not always apply to instances of the class

I do rather favor the separation of these concerns, and the rate seems somewhat important, so I'd rather have a separate class `PulsedExcitationSource` that requires the pulse rate